test/nodetool/test_nodetool.py

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import pytest
+import rest_api_mock
+import json
+import subprocess
+
+
+def test_jmx_compatibility_args(server_context, nodetool):
+    """Test that JMX related command line arguments are ignored.
+
+    Since we cannot invoke nodetool without a concrete command, we use nodetool
+    compact here, but this is irrelevant, we are just checking that JMX-related
+    command-line args are recognized and ignored.
+    """
+    rest_api_mock.set_expected_requests(server_context, [
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema")])
+
+    nodetool("compact", "system_schema", "-u", "us3r", "-pw", "secr3t")
+    nodetool("compact", "system_schema", "--username", "us3r", "--password", "secr3t")
+    nodetool("compact", "system_schema", "-u", "us3r", "-pwf", "/tmp/secr3t_file")
+    nodetool("compact", "system_schema", "--username", "us3r", "--password-file", "/tmp/secr3t_file")
+    nodetool("compact", "system_schema", "-pp")
+    nodetool("compact", "system_schema", "--print-port")
```

Comment:
A thought (you don't have to do it now):

Is there any performance advantage to listing all the expected requests for 10 different nodetool commands, instead of doing the requests one by one and expecting the outputs for each one separate? If not, you could have had a test function like
```python
nodetool_expect("compact", "system_schema", "--print-port", [("POST", "/storage_service/keyspace_compaction/system_schema")])
```
This nodetool_expect will run a single nodetool command and expects a list of REST API calls (and insists that exactly this list, not less and not more, is what happens).

Reply:
Good idea, the current system is indeed hard to follow, as soon as you have more than 1-2 requests in a test. I will implement your suggestion, it shouldn't be hard or have any noticeable performance penalties. Furthermore, it will improve isolation, breakages will be discovered earlier.

Reply:
Done in v4.